src/mgmt/rpc/server/IPCSocketServer.cc

```diff
@@ -219,11 +217,12 @@ IPCSocketServer::run()
 
     std::error_code ec;
     if (int fd = this->accept(ec); !ec) {
-      Client client{fd};
+      Client client{fd, _conf.incomingRequestMaxBufferSize};
+      Buffer bw;
 
       if (auto [ok, errStr] = client.read_all(bw); ok) {
-        const auto   json = std::string{bw.data(), bw.size()};
-        rpc::Context ctx;
+        const std::string &json = bw.str();
```

Comment:
I think this could create a temporary and a copied string.  I suggest having str() return string_view and storing it by value here.

Reply:
I did the `const &` intentionally, to extend(from the `str()` return by value)  the lifetime of the temporary string returned by `str()` so we just 1  copy which will be stored into `json`.

I can do either way, no strong feeling on any one.

Reply:
I understand extending the lifetime, but a copy seems unnecessary since the buffer is already managed by the Buffer class.   I think Buffer::str could just call flush() and then return a const ref to the internal string and you'd get the same allocations for the small case, but save a copy when reading > N bytes.  Its not a huge deal based on how often this happens.

Reply:
no it's fine, it should be just 1 ideally, that was the point at least.